### DISCUSSION TITLE
Seq EULA/Licensing Inquiry

### DISCUSSION THREAD

**Discussion by svengeance** [datalust/seq-tickets#1172] [Help]
Hey @nblumhardt, thanks for the wonderful product!

I'm looking at incorporating Seq into a test framework I'm building, and I would like your guidance on whether or not I am within the licensing agreement for Seq. This will be an open-source extension on NUnit's Test Adapter. What I would like to do is _containerize_ each test in a test class. To communicate to the user what is happening inside the container, I would like to use Serilog and output to Seq.

I want to make sure this is within Seq's EULA. The intent is that my test adapter would create a single seq container on the user's machine, and the user can navigate to the forwarded port on the host machine to see the output of their tests in real time. I would be likewise directing the user to Seq's license agreement.

Please let me know your thoughts!

**Comment by nblumhardt** [comment]
Hi Sven, thanks for getting in touch - it sounds like a very interesting project.

Unfortunately it's difficult to quickly think through any potential issues/implications, so I can't offer guidance on this just now - having the user either download and install Seq from the website, or run the container themselves to get output into Seq, might be a reasonable workflow in the interim?

Best regards,
Nick

**Reply by svengeance** [reply]
This will certainly work in the interim, thank you. I will provide instruction but _not_ the automation of standing up Seq containers.

If you could put some thought into allowing this automation I would love to hear back from you. In this scenario I would like to encourage developers to use Seq for local development and testing, as seq's interaction with structured logging is bar-none.

Thank you very much!
Sven

**Reply by svengeance** [reply]
Also - I will shortly (hopefully end of week) be able to provide a video demonstration of what I mean to accomplish if that would help potentially persuade you 😉 

I will post another reply when I can show you.

**Reply by svengeance** [reply]
@nblumhardt Here is a complete picture - https://gfycat.com/MajesticSpiritedEyas

You can see that the action of running tests will create a Seq container for users to see what's happening inside the container.

Given that there's no interaction with users in this workflow, I'll likely have to provide outside instruction (e.g. "if you have this container open on these ports connected to this network, logs will be output to Seq!").

I don't expect this to change your mind today, but I hope to provide you a full understanding of what I would like to automate. For now I can use something like https://github.com/mthenw/frontail in an automated fashion, but I would love to stand up Seq from the get-go.

Thanks again!
